### PR TITLE
Feature/support https rpc client

### PR DESCRIPTION
**Pull request by ruseinov**
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

File: rpc/lib/client/http_client.go [hunk 1]
```diff
@@ -25,7 +25,10 @@ type HTTPClient interface {
 }
 
 // TODO: Deprecate support for IP:PORT or /path/to/socket
-func makeHTTPDialer(remoteAddr string) (string, func(string, string) (net.Conn, error)) {
+func makeHTTPDialer(remoteAddr string) (string, string, func(string, string) (net.Conn, error)) {
```
Comment (ethanfrey) [inline rpc/lib/client/http_client.go:36]: Seems like the simplest solution, to add the http/https protocol here.
This is not used anywhere else than the below points, right? (Being private doesn't seem a bad change).

File: rpc/lib/client/ws_client.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ type WSClient struct {
 // functions for a detailed description of how to configure ping period and
 // pong wait time. The endpoint argument must begin with a `/`.
 func NewWSClient(remoteAddr, endpoint string, options ...func(*WSClient)) *WSClient {
-	addr, dialer := makeHTTPDialer(remoteAddr)
+	_, addr, dialer := makeHTTPDialer(remoteAddr)
```
Comment (ethanfrey) [inline rpc/lib/client/ws_client.go]: This doesn't seem to differentiate between ws:// and wss://, which should be equivalent to the difference between http and https.

I had to specify that to get the connection working in javascript: https://github.com/confio/weave-js/blob/master/src/tendermint/rpc.js#L27-L46
maybe something similar is needed here.

**Review by ethanfrey — APPROVED** [review]
Looks like a good change.
Can you please test the websocket connections over ssl (wss://) to verify it works?

**Comment by ruseinov** [comment]
@alpe @ethanfrey done the wss change.

File: rpc/lib/client/ws_client.go [hunk 3]
```diff
@@ -242,7 +251,7 @@ func (c *WSClient) dial() error {
 		Proxy:   http.ProxyFromEnvironment,
 	}
 	rHeader := http.Header{}
-	conn, _, err := dialer.Dial("ws://"+c.Address+c.Endpoint, rHeader)
+	conn, _, err := dialer.Dial(c.protocol+"://"+c.Address+c.Endpoint, rHeader)
```
Comment (ethanfrey) [inline rpc/lib/client/ws_client.go:254]: nice job

**Comment by ethanfrey** [comment]
Great, if you tested this against a running wss/https service, then great!
I approve the code, but never ran it

**Comment by Isabello** [comment]
Are you sure merging that 250 commits to this PR was correct? @ruseinov

**Comment by ruseinov** [comment]
@Isabello disregard this PR, I have opened another one directly to tendermint. Closing this.

**Pull request closed without merging**